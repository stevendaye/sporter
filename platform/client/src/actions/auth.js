import { REGISTER_SUCCESS, REGISTER_FAIL, USER_LOADED, AUTH_TOKEN_ERROR,
  LOGIN_SUCCESS, LOGIN_FAIL, LOGOUT
} from "../constants/types";

// Register Action Creators
const doRegisterSuccess = payload => ({
  type: REGISTER_SUCCESS,
  payload
});

const doRegisterFail = () => ({
  type: REGISTER_FAIL
});

// Login User Action Creators
const doLoginSuccess = payload => ({
  type: LOGIN_SUCCESS,
  payload
});

const doLoginFail = () => ({
  type: LOGIN_FAIL
});


// Load User Info Action Creators
const doLoadUser = payload => ({
  type: USER_LOADED,
  payload
});

const doSetAuthError = () => ({
  type: AUTH_TOKEN_ERROR
});

// Logout user
const doLogout = () => ({
  type: LOGOUT
});

export { doRegisterSuccess, doRegisterFail, doLoadUser, doSetAuthError,
  doLoginSuccess, doLoginFail, doLogout
};
