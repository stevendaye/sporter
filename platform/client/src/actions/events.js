import { EVENTS_STORE, EVENTS_GET, EVENT_VOTE, EVENTS_ERROR } from "../constants/types";

const doStoreEvents = payload => ({
  type: EVENTS_STORE,
  payload
});

const doSetEvents = payload => ({
  type: EVENTS_GET,
  payload
});

const doVoteEventSuccess = payload => ({
  type: EVENT_VOTE,
  payload
});

const doEventActionFail = () => ({
  type: EVENTS_ERROR
});

export { doStoreEvents, doSetEvents, doVoteEventSuccess, doEventActionFail };
