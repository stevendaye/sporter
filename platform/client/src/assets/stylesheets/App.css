/* #### Sports Poll Main Stylesheet #### */

/* Navbar & Header Styles */
.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: .6rem 3.5rem;
  position: fixed;
  z-index: 1;
  width: 100%;
  top: 0;
  opacity: 1;
}

.navbar ul {
  display: flex;
  left: 0;
  margin: 0;
  padding: 0;
}

.navbar a {
  color: #fff;
  padding: 0.45rem;
  margin: 0 0.25rem;
}

.navbar a:hover {
  color: inherit;
}

.navbar .welcome span {
  margin-right: 0.6rem;
}

/* Container tyles */
.wrapper {
  position: relative;
  min-height: 100vh;
}

.container {
  max-width: 900px;
  margin: 0 auto;
  overflow: hidden;
  padding: 0 2rem;
  margin-top: 6rem;
}

.wrap-set-bottom {
  margin-bottom: 5rem;
}

/* Footer Styles */
.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 2.5rem;
  padding: 0 25px;
  margin: 2rem 0 0.5rem 0;
  color: var(--dim-color);
  font-size: .8em;
  text-align: center;
}

.footer a {
  color: var(--dim-color);
}