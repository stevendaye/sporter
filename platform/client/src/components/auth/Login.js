import React, { Fragment, Component } from "react";
import { Link, Redirect } from "react-router-dom"
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Notifications from "../../components/layout/Notifications";
import doLoginWithErrorCheck from "../../thunks/login";
import facebookLogo from "../../assets/vendor/facebook/facebook_logo_white.png";
import twiiterLogo from "../../assets/vendor/twitter/twitter_logo_white.png";

class Login extends Component {
  _isMounted = false;

  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      submitted: false
    }

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const { feedback } = this.props;
    const { email, password } = this.state;
    this.props.onLogin(email, password);

    if (email !== "" && email.includes("@") && password.length >= 6) {
      this.setState({ submitted: true });
    }

    if (feedback === "Invalid Entries") {
      this.setState({ submitted: false });
    } else if (feedback === "Valid Entries") {
      this.setState({ submitted: true });
    }
  }
  UNSAFE_componentWillReceiveProps(props) {
    if (props.feedback === "Invalid Entries") {
      this.setState({ submitted: false });
    } else if (props.feedback === "Valid Entries") {
      this.setState({ submitted: true });
    }
  }

  render() {
    const { isAuthenticated } = this.props;
    const { email, password, submitted } = this.state;

    if (isAuthenticated) {
      return <Redirect to = "/events"/>
    }

    return (
      <Fragment>
        <div className = "bg-img">
          <div className = "dark-overlay">
            <div className = "form-sign">
              <p className = "lead text-white"> Log into your account</p>
              <Notifications/>

              <form className = "form" onSubmit={ this.onSubmit }>
                <div className = "form-group">
                  <input
                    type = "email"
                    placeholder = "Email Address"
                    name = "email"
                    value = {email}
                    onChange = { this.onChange }
                    required
                  />
                </div>
                <div className = "form-group">
                  <input
                    type = "password"
                    placeholder = "Password"
                    name = "password"
                    minLength = "6"
                    value = {password}
                    onChange = { this.onChange }
                    required
                  />
                </div>
                <input
                  type = {submitted ? "button" : "submit"}
                  style = {{ width: "100%" }}
                  className = {submitted ? "btn btn-dim" : "btn btn-primary"}
                  value = {submitted ? "Loging in..." : "Login"}
                />
              </form>
              <div style = {{ margin: "25px 0 0 0" }} className = "social-login text-white">
                <div className = "login-with">
                  <hr/>
                  <span> Or With </span>
                  <hr/>
                </div>
                <a href = "/users/auth/login/facebook">
                  <button style = {{ margin: "15px 0 0 0", display: "flex", justifyContent: "center" }} className = "fa-facebook">
                    <div style = {{ border: "none", width: "20px", height: "20px", margin: "0.5px 2px 0 0" }}>
                      <img src = {facebookLogo} alt = "" title = "" />
                    </div>
                    facebook
                  </button>
                </a>
                <a href = "/users/auth/login/twitter">
                  <button style = {{ margin: "15px 0 0 0", display: "flex", justifyContent: "center" }} className = "fa-twitter">
                    <div style = {{ border: "none", width: "20px", height: "20px", margin: "2px 7px 0 0" }}>
                      <img src = {twiiterLogo} alt = "" title = "" />
                    </div>
                    twitter
                  </button>
                </a>
              </div>
              <br/>
              <p className = "my-1 text-white">
                Don't have an account? <Link to = "/register">Register</Link>
              </p>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

Login.propTypes = {
  isAuthenticated: PropTypes.bool,
  feedback: PropTypes.string,
  onLogin: PropTypes.func.isRequired
}

const mapStateToPropsLogin = state => ({
  isAuthenticated: state.authState.isAuthenticated,
  feedback: state.entriesState.message
});

const mapDispatchToPropsLogin = dispatch => ({
  onLogin: (email, password) =>
    dispatch(doLoginWithErrorCheck(email, password))
});

const ConnectedLogin = connect(mapStateToPropsLogin, mapDispatchToPropsLogin)(Login);

export default ConnectedLogin;
