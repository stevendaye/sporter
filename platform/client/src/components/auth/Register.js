import React, { Fragment, Component } from "react";
import { Link, Redirect } from "react-router-dom"
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Notifications from "../../components/layout/Notifications";
import doSetRemoveNotifications from "../../thunks/notifications";
import doRegisterWithErrorCheck from "../../thunks/register";

class Register extends Component {
  _isMounted = false;

  constructor (props) {
    super(props);

    this.state = {
      name: "",
      email: "",
      password: "",
      confirmPass: "",
      submitted: false,
    }

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const { feedback } = this.props;
    const { name, email, password, confirmPass } = this.state;
    password !== confirmPass
    ? this.props.onSetNotification("Passwords do not match!", "danger")
    : this.props.onRegister(name, email, password);

    if (password === confirmPass && password.length >= 6 && name.length > 5 && email.includes("@")) {
      this.setState({ submitted: true });
    }

    if (feedback === "Invalid Entries") {
      this.setState({ submitted: false });
    } else if (feedback === "Valid Entries") {
      this.setState({ submitted: true });
    }
  }

  UNSAFE_componentWillReceiveProps(props) {
    if (props.feedback === "Invalid Entries") {
      this.setState({ submitted: false });
    } else if (props.feedback === "Valid Entries") {
      this.setState({ submitted: true });
    }
  }

  render() {
    const { isAuthenticated } = this.props;
    const { name, email, password, confirmPass, submitted } = this.state;

    if (isAuthenticated) {
      return <Redirect to = "/events" />
    }

    return (
      <Fragment>
        <div className = "bg-img">
          <div className = "dark-overlay">
            <div className = "form-sign">
              <p className = "text-white lead">Create an account</p>
              <Notifications/>

              <form className = "form" onSubmit = { this.onSubmit }>
                <div className = "form-group">
                  <input
                    type = "text"
                    placeholder = "Name"
                    name = "name"
                    value = {name}
                    onChange = { this.onChange }
                    ref = { el => this.nameInput = el }
                    required
                  />
                </div>
                <div className = "form-group">
                  <input
                    type = "email"
                    placeholder = "Email Address"
                    name = "email"
                    value = {email}
                    onChange = { this.onChange }
                    required
                  />
                </div>
                <div className = "form-group">
                  <input
                    type = "password"
                    placeholder = "Password"
                    name = "password"
                    minLength = "6"
                    value = {password}
                    onChange = { this.onChange }
                    required
                  />
                </div>
                <div className = "form-group">
                  <input
                    type = "password"
                    placeholder = "Confirm Password"
                    name = "confirmPass"
                    minLength = "6"
                    value = {confirmPass}
                    onChange = { this.onChange }
                    required
                  />
                </div>
                <input
                  type = {submitted ? "button" : "submit"}
                  style = {{ width: "100%" }}
                  className = { submitted ? "btn btn-dim" : "btn btn-primary"}
                  value = {submitted ? "Registering..." : "Register"}
                />
              </form>
              <br/>
              <p className = "my-1 text-white">
                Already have an account? <Link to="/login">Login</Link>
              </p>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

Register.propTypes = {
  feedback: PropTypes.string,
  isAuthenticated: PropTypes.bool,
  onRegister: PropTypes.func.isRequired,
  onSetNotification: PropTypes.func.isRequired
}

const mapStateToPropsRegister = state => ({
  isAuthenticated: state.authState.isAuthenticated,
  feedback: state.entriesState.message
});

const mapDispatchToPropsRegister = dispatch => ({
  onSetNotification: (message, alert) =>
    dispatch(doSetRemoveNotifications(message, alert)),
  onRegister: (name, email, password) =>
    dispatch(doRegisterWithErrorCheck(name, email, password))
});

const ConnectedRegister = connect(mapStateToPropsRegister, mapDispatchToPropsRegister)(Register);

export default ConnectedRegister;
