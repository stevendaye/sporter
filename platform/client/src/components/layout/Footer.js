import React, { Fragment } from "react";
import { Link } from "react-router-dom";

const Footer = () =>
  <Fragment>
    <footer className = "footer">
      <Link to = "#/policy">Privacy &amp; Policy</Link> &nbsp;&nbsp;&nbsp;
      <Link to = "#/feedback">Feedback &amp; Suggestions</Link> &nbsp;&nbsp;&nbsp;
      &copy; 2019 Sports Poll, All Right Reserved
    </footer>
  </Fragment>

export default Footer;
