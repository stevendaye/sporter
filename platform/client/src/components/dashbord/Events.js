import React, { Fragment, Component } from "react";
import { connect } from "react-redux";
import Moment from "react-moment";
import PropTypes from  "prop-types";
import Notifications from "../layout/Notifications";
import { doVoteEventWithErrorCheck } from "../../thunks/events";
import { doGetEventsWithErrorCheck } from "../../thunks/events";
import doSetRemoveNotification from "../../thunks/notifications";
import avatar from "../../assets/vendor/local/avatar.png";

class Events extends Component {
  _isMounted = false;
  constructor(props) {
    super(props);

    this.state = {
      event: null,
      quickRefresh: true,
      onceVoted: false
    }

    this.onVote = this.onVote.bind(this);
    this.onClick = this.onClick.bind(this);
    this.onSetEvent = this.onSetEvent.bind(this);
    this.onCheck = this.onCheck.bind(this);
  }

  componentDidMount() {
    this._isMounted = true

    /* Generate a random event */
    const { events, auth: { user } } = this.props;
    const random = Math.random(), len = events.length;
    const event = events[Math.floor(random * len)];
    this.setState({ event });

    /* Check if the logged in user voted the current random event */
    this.state.event && (JSON.parse(this.state.event.votes)
    .filter(vote => vote.userid === user.id).length > 0)
    && this.setState({ onceVoted: true });

    /* Get all list of events at mount for the sake of the
    UNSAFE_componentWillReceiveProps component lifecycle method */
    this._isMounted && this.props.onGetEvents();
  }

  UNSAFE_componentWillReceiveProps() {
    const { quickRefresh } = this.state;
    const { events } = this.props;
    const len = events.length;
    if (quickRefresh) {
      if (events && len > 0) {
        const random = Math.random();
        const current = events[Math.floor(random * len)];
        this.onSetEvent(current);
      }
    }
  }

  onSetEvent(current) {
    this.setState({ event: current });
  }

  onClick(timeout) {
    setTimeout(() => {
      const { events } = this.props;
      const random = Math.random(), len = events.length;
      const event = events[Math.floor(random * len)];
      this.setState({ event });
    }, timeout);
  }

  onCheck(objectId, poll) {
    const { event } = this.state;
    const { auth: { isLoading, user } } = this.props;
    if (!isLoading &&
    JSON.parse(event.votes).filter(vote => vote.userid === user.id).length > 0) {
      this.props.onPushNotification();
      this.setState({ onceVoted: true });
    } else {
      this.onVote(objectId, poll);
      this.setState({ onceVoted: false });
      this.setState({ quickRefresh: false });
    }
    this.onClick(1500);
  }

  onVote(objectId, poll) {
    this.props.onSendVote(objectId, poll);
  }

  render() {
    const { event } = this.state;
    const { auth: { user } } = this.props;

    return (
      <Fragment>
        {!event || !user
          ? <h3 className = "text-dim block-center f-2">Loading Event...</h3>
          : (
            <section style = {{ position: "relative", margin: "0" }}>
              <Notifications/>
              <h2 className = "text-center" style = {{ margin: "0" }}>Sport Events</h2>
              <p style = {{ float: "right", cursor: "pointer" }}
                className = "text-primary" onClick = { () => { this.onClick(100) } }
              >
                Skip <span role = "img" aria-label = "img">&#9193;</span>
              </p>
              <div className = "clear"></div>
              <div className = "badge">
                <div className = "text-center pxy-1 px-2 my-2 bg-dim b-r-3">
                  {event.name} - <span className = "text-primary">{event.state}</span>
                </div>
                <div className = "text-center f-1 text-primary">
                  GROUP: {(event.group).toUpperCase()}
                </div>
                <div className = "badge-content">
                  <div className = "badge-content-1">
                    <div className = "badge-img-size">
                      <img className = "round-img img-avatar" src = {avatar}
                        alt = "Avatar" width = "100%" height = "100%"
                      />
                    </div>
                    <p style = {{ fontSize: "1.2rem" }} >{event.homeName}</p>
                    <p className = "text-primary f-1"><strong>HOME</strong></p>
                  </div>

                  <div className = "badge-content-2">
                    <span className = "f-3" role = "img" aria-label = "img">&#9918;</span>
                    <p className = "bg-dim px-2 b-r-6">{event.sport}</p>
                    <p className = "text-dark f-3">{event.country}</p>
                    <p className = "text-dim f-1">
                      Posted: <Moment format = "DD/MM/YYYY">{event.createdAt}</Moment>
                    </p>
                  </div>

                  <div className = "badge-content-3">
                    <div className = "badge-img-size">
                      <img className = "round-img img-avatar" src = {avatar}
                        alt = "Avatar" width = "100%" height = "100%"
                      />
                    </div>
                    <p style = {{ fontSize: "1.2rem" }} >{event.awayName}</p>
                    <p className = "text-primary f-1"><strong>AWAY</strong></p>
                  </div>
                </div>

                <div className = "btn-actions">
                  <button className = {
                    JSON.parse(event.votes)
                    .filter(vote => vote.userid === user.id && vote.poll === "home-win").length > 0
                    ? "btn-voted btn-grow"
                    : "btn-vote btn-grow" }
                    onClick = { () => {
                      this.onCheck(event.objectId, "home-win");
                      this.xbtn.className = "btn-vote btn-grow";
                    } }
                    ref = { el => this.xbtn = el }
                  >
                    Home Win
                  </button>

                  <button className = {
                    JSON.parse(event.votes)
                    .filter(vote => vote.userid === user.id && vote.poll === "draw").length > 0
                    ? "btn-voted btn-grow"
                    : "btn-vote btn-grow" }
                    onClick = { () => {
                      this.onCheck(event.objectId, "draw");
                      this.ybtn.className = "btn-vote btn-grow";
                    } }
                    ref = { el => this.ybtn = el }
                  >
                    Draw Game
                  </button>

                  <button className = {
                    JSON.parse(event.votes)
                    .filter(vote => vote.userid === user.id && vote.poll === "away-win").length > 0
                    ? "btn-voted btn-grow"
                    : "btn-vote btn-grow" }
                    onClick = { () => {
                      this.onCheck(event.objectId, "away-win")
                      this.zbtn.className = "btn-vote btn-grow";
                    } }
                    ref = { el => this.zbtn = el }
                  >
                    Away Win
                  </button>
                </div>
              </div>
            </section>
          )
        }
      </Fragment>
    );
  }
};

Events.propTypes = {
  auth: PropTypes.object.isRequired,
  events: PropTypes.array.isRequired,
  onPushNotification: PropTypes.func.isRequired,
  onGetEvents: PropTypes.func.isRequired,
  onSendVote: PropTypes.func.isRequired
};

const mapStateToPropsEvents = state => ({
  auth: state.authState,
  events: state.eventsState.events
});

const mapDispatchToPropsEvents = dispatch => ({
  onPushNotification: () =>
    dispatch(doSetRemoveNotification("YOU CAN ONLY VOTE ONCE. TRY NEXT OR SKIP!", "failure")),
  onSendVote: (objectId, poll) =>
    dispatch(doVoteEventWithErrorCheck(objectId, poll)),
  onGetEvents: () =>
    dispatch(doGetEventsWithErrorCheck())
});

const ConnectedEvents = connect(mapStateToPropsEvents, mapDispatchToPropsEvents)(Events);

export default ConnectedEvents;
