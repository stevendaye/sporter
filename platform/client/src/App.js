import React, { Fragment, useEffect } from 'react';
import { BrowserRouter as Router } from "react-router-dom";
import { Route, Switch } from "react-router";
import { Provider } from "react-redux";
import store from "./store";
import Routes from "./components/routes/Routes";
import Navbar from "./components/layout/Navbar";
import Landing from "./components/layout/Landing";
import Login from "./components/auth/Login";
import Register from "./components/auth/Register";
import Footer from "./components/layout/Footer";
import setAuthTokenHeader from "./utils/setAuthTokenHeader";
import { doLoadUserWithErrorCheck } from "./thunks/register";
import doStoreEventsWithErrorCheck, { doGetEventsWithErrorCheck } from "./thunks/events";
import "./assets/stylesheets/App.css";

if (localStorage.token) {
  setAuthTokenHeader(localStorage.token);
}

const App = () => {
  useEffect(() => {
    store.dispatch(doLoadUserWithErrorCheck());
    store.dispatch(doStoreEventsWithErrorCheck());
    store.dispatch(doGetEventsWithErrorCheck());
  }, []);

  return (
    <Provider store = {store}>
      <Router>
        <Fragment>
          <div className = "wrapper">
            <Navbar/>
            <Switch>
              <Route exact path = "/" component = { Landing } />
              <Route exact path = "/login" component = { Login } />
              <Route exact path = "/register" component = { Register } />
              <Route component = { Routes } />
            </Switch>
            <Footer/>
          </div>
        </Fragment>
      </Router>
    </Provider>
  );
};

export default App;
