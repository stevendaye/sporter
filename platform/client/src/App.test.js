import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import deepFreeze from "deep-freeze";
import App from './App';
import Navbar from "./components/layout/Navbar";
import Landing from "./components/layout/Landing";
import Footer from "./components/layout/Footer";
import Notifications from "./components/layout/Notifications";
import NotFound from "./components/layout/NotFound";
import Register from "./components/auth/Register";
import Login from "./components/auth/Login";
import Events from "./components/dashbord/Events";
import Routes from "./components/routes/Routes";
import PrivateRoute from "./components/routes/PrivateRoute";
import { doStoreEvents, doSetEvents } from "./actions/events";
import { doRegisterSuccess, doLoadUser } from "./actions/auth";
import eventsReducer from "./reducers/events";
import authReducer from "./reducers/auth";

/* ## Create test suites for testing components ## */
/* React Components */
describe("App", () => {
  // Test rendition of component
  it("renders without crashing", ()=> {
    const div = document.createElement('div');
    ReactDOM.render(
      <App />, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  // Test snapshot for future tests
  it("has a valid snapshot", done => {
    const component = renderer.create(<App/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Navabar", () => {
  it("renders without crashing", done => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Navbar/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has valid snapshot", done => {
    const component = renderer.create(<Navbar/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Landing", ()=> {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Landing/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has valid snapshot", done => {
    const component = renderer.create(<Navbar/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Footer", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Footer/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has valid snapshot", done => {
    const component = renderer.create(<Footer/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Register", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Register/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has valid snapshot", done => {
    const component = renderer.create(<Register/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Login", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Login/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has valid snapshot", done => {
    const component = renderer.create(<Login/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Notifications", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Notifications/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", done => {
    const component = renderer.create(<Notifications/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("NotFound", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <NotFound/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", done => {
    const component = renderer.create(<NotFound/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Events", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Events/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", done => {
    const component = renderer.create(<Events/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done()
  });
});

describe("Routes", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Routes/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", done => {
    const component = renderer.create(<Routes/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("PrivateRoute", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <PrivateRoute/>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", done => {
    const component = renderer.create(<PrivateRoute/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});


/* ## Create Test suites for testing Reducers and Action Creators and Selectors ## */
/* Action Creators */
describe("Store Events Action", () => {
  it("stores events", done => {
    const events = [{
      awayName: "Panthrakikos Komotini",
      createdAt: "2015-12-18T12:30:39.228Z",
      group: "Greek Cup",
      homeName: "Chania FC",
      id: 1002916450,
      name: "Chania FC - Panthrakikos Komotini",
      objectId: "1UaQjc7lIb",
      sport: "FOOTBALL",
      country: "ENGLAND",
      state: "STARTED"
    }];

    const action = doStoreEvents(events);
    const expectation = {
      type: "EVENTS_STORE",
      events
    };
    expect(action).toStrictEqual(expectation);
    done();
  });
});

describe("Get Events Action", () => {
  it("gets events", done => {
    const events = [{
      awayName: "Panthrakikos Komotini",
      createdAt: "2015-12-18T12:30:39.228Z",
      group: "Greek Cup",
      homeName: "Chania FC",
      id: 1002916450,
      name: "Chania FC - Panthrakikos Komotini",
      objectId: "1UaQjc7lIb",
      sport: "FOOTBALL",
      country: "ENGLAND",
      state: "STARTED"
    }];

    const action = doSetEvents(events);
    const expectation = {
      type: "EVENTS_GET",
      events
    };
    expect(action).toStrictEqual(expectation);
    done();
  });
});

describe("Register User Action", () => {
  it("register user", done => {
    const token = "V6jd702hdNS673HldG6S2fkK92HS692HV642Lhkd3hdh394";
    const action = doRegisterSuccess(token);
    const expectation = {
      type: "REGISTER_SUCCESS",
      token
    };
    expect(action).toStrictEqual(expectation);
    done();
  });
});

describe("Load User Action", () => {
  it("loads user", done => {
    const user = {
      avatar: "//www.gravatar.com/avatar/4d73599996d6e6a4a84c88fc6f6ebc13?s=200&r=pg&d=mm",
      email: "elienoma@yahoo.com",
      id: "84e3979e-fa1a-40f0-b951-7ee3e7aa275f",
      name: "Elie Noma",
      provider: "local",
      timestamp: "2019-12-04T18:15:22.187Z"
    };
    const action = doLoadUser(user);
    const expectation = {
      type: "USER_LOADED",
      user
    };
    expect(action).toStrictEqual(expectation);
    done();
  });
});

/* Reducers */
describe("Events Reducer", () => {
  it("populates events in the events state", done => {
    const events = [{
      awayName: "Panthrakikos Komotini",
      createdAt: "2015-12-18T12:30:39.228Z",
      group: "Greek Cup",
      homeName: "Chania FC",
      id: 1002916450,
      name: "Chania FC - Panthrakikos Komotini",
      objectId: "1UaQjc7lIb",
      sport: "FOOTBALL",
      country: "ENGLAND",
      state: "STARTED"
    }];
    const prevState = {
      events: [],
      event: null,
      isLoading: true,
      message: ""
    };
    const esxpectedNewState = {
      events,
      event: null,
      isLoading: true,
      message: ""
    };
    const action = {
      type: "EVENTS_STORE",
      events
    };
    deepFreeze(prevState);

    const newState = eventsReducer(prevState, action);
    expect(newState).toStrictEqual(esxpectedNewState);
    done();
  });
});

describe("Auth Reducer", () => {
  it("puts user credentials in the auth state", done => {
    const user = {
      avatar: "//www.gravatar.com/avatar/4d73599996d6e6a4a84c88fc6f6ebc13?s=200&r=pg&d=mm",
      email: "elienoma@yahoo.com",
      id: "84e3979e-fa1a-40f0-b951-7ee3e7aa275f",
      name: "Elie Noma",
      provider: "local",
      timestamp: "2019-12-04T18:15:22.187Z"
    };
    const prevState = {
      user: null,
      token: "V6jd702hdNS673HldG6S2fkK92HS692HV642Lhkd3hdh394",
      isAuthenticated: true,
      isLoading: false
    };
    const action = {
      type: "USER_LOADED",
      user
    };
    const esxpectedNewState = {
      user: {
        avatar: "//www.gravatar.com/avatar/4d73599996d6e6a4a84c88fc6f6ebc13?s=200&r=pg&d=mm",
        email: "elienoma@yahoo.com",
        id: "84e3979e-fa1a-40f0-b951-7ee3e7aa275f",
        name: "Elie Noma",
        provider: "local",
        timestamp: "2019-12-04T18:15:22.187Z"
      },
      token: "V6jd702hdNS673HldG6S2fkK92HS692HV642Lhkd3hdh394",
      isAuthenticated: true,
      isLoading: false
    };
    deepFreeze(prevState);

    const newState = authReducer(prevState, action);
    expect(newState).toStrictEqual(esxpectedNewState);
    done();
  });
});

/* Selectors */
// Selectors test goes here
