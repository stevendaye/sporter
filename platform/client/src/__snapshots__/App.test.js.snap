// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="wrapper"
>
  <nav
    className="navbar bg-dark"
  >
    <h1>
      <a
        href="/"
        onClick={[Function]}
      >
        Sports Poll
      </a>
    </h1>
  </nav>
  <section
    className="landing"
  >
    <div
      className="dark-overlay"
    >
      <div
        className="landing-inner"
      >
        <h1
          className="large"
        >
          The true champion believes in the impossible
        </h1>
        <p
          className="lead"
        >
          VOTE LIKE A CHAMPION TODAY!
        </p>
        <div
          className="buttons"
        >
          <a
            className="btn btn-primary hide-sm show-md"
            href="/events"
            onClick={[Function]}
          >
            Start voting your team
          </a>
          <a
            className="btn btn-primary show-sm hide-default"
            href="/register"
            onClick={[Function]}
          >
            Register Now
          </a>
          <br />
           
          <br />
          <a
            className="btn btn-primary show-sm hide-default"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <a
      href="/#/policy"
      onClick={[Function]}
    >
      Privacy & Policy
    </a>
        
    <a
      href="/#/feedback"
      onClick={[Function]}
    >
      Feedback & Suggestions
    </a>
         © 2019 Sports Poll, All Right Reserved
  </footer>
</div>
`;

exports[`NotFound has a valid snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1
    className="x-large text-primary"
  >
    <span
      style={
        Object {
          "display": "block",
        }
      }
    >
       ☹ 
    </span>
    Page Not Found
  </h1>
  <p
    className="large"
  >
    Sorry, you have taken a wrong turn
  </p>
</div>
`;
